rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,59 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/) (#community:nixos.org) as our official
+community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space (#community:nixos.org) that has already been created which you
+can join by visting the [Invite
+link](https://app.element.io/#/room/#community:nixos.org).
+
+Update any other places that point to Freenode to point to Matrix
+instead.
+
+Port any bots that might be necessary such as:
+
+* nix-channel-monitor
+* github PR events
+* karma bot
+* eval bot
+* factoids bot
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+People are used to Freenode and it might be harder for people to use
+Matrix, which requires more complex clients.
+
+# Alternatives
```

Comment:
Which always results in degradations of chat quality on one side, which is why I would be very careful about bridging in general.

Reply:
I think there's a difference between "Matrix user uses a bridge to talk in an IRC channel", vs. "the community is on Matrix but an IRC interface is provided through a bridge".

In the former case, the baseline to compare against is "only native IRC users", whereas in the latter case the baseline is "no access from IRC at all". So long as the messaging around this is clear, I think that imperfect IRC access is better than no IRC access.

Reply:
The more I think about it, the more I feel that the RFC should at the very least *mention* IRC bridging.

I think it's desirable to have this (with the understanding of it being a Matrix-first room, feature/convention-wise), not just to improve accessibility for a certain part of the community, but also because "my system won't boot and all I have is netcat" is a realistic problem for someone trying to access a distro room to have, and IRC can work in this scenario.

Libera does not currently have an operational bridge, but likely will in the future. OFTC already has an operational bridge today.